core/src/dataSync.ts

```diff
@@ -0,0 +1,44 @@
+// backend/src/dataSync.ts
+import { Client, Message } from 'whatsapp-web.js';
+import { userConfig, saveUserConfig } from './dataStore.js';
+import { handleSelfChatCommand } from './commandHandler.js';
+
+export async function syncDataFromSelfChatHistory(client: Client) {
+	console.log('Starting data sync from self-chat history...');
+	try {
+		const chats = await client.getChats();
+		const selfChat = chats.find(chat => chat.id.user === client.info.wid.user && !chat.isGroup); // Find your own chat
```

Comment:
We should also be looking at the messages sent in the dedicated command chat if the user has that set in their preferences.